### PR TITLE
[constexpr-contracts] Adds new port

### LINKED ISSUE #11578: [New Port Request] constexpr-contracts
Library name: constexpr-contracts

Library description: "Pre-conditions, post-conditions, and assertions, all available at compile-time "

Source repository URL: https://github.com/cjdb/constexpr-contracts

### PR DESCRIPTION
**Pull request by cjdb**
**Describe the pull request**

adds constexpr-contracts to vcpkg

> What does your PR fix?

Fixes #11578 

> Which triplets are supported/not supported? Have you updated the CI baseline?

All triplets should be supported if the compilers used are recent enough.

> Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

> Make separate Pull Requests per port

Yes.

> Avoid trivial changes in untouched files

Yes.

> Check names against other repositories

`constexpr-contracts` not currently identified by Repology.

> c000001: No deprecated helper functions are used

No.

> c000002: Control Version field is updated

Yes.

> c000003: New ports contain a Description field written in English

Yes.

> c000004: No unnecessary comments are present in the changeset

No.

> c000005: Downloaded archives are versioned if available

**Unsure.**

> c000006: New ports pass CI checks for triplets that the library officially supports

**Should be in progress upon PR submission.**

> c000007: Patches fix issues that are vcpkg-specific only

Yes.

> c000008: New ports download source code from the official source if available

I am the source author and can confirm this is true.

> c000010: Ports and port features are named correctly

Yes.

> c000011: Library targets are exported when appropriate

**Unsure.**

> c000012: Ports do not use applications which modify the user's system

Yes.

> c000013: Ports with system dependencies include an information message during installation

No dependencies beyond a C++20 compiler.

### CODE REVIEW

File: ports/constexpr-contracts/CONTROL [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Source: constexpr-contracts
+Version: 2020-05-25
+Homepage: https://github.com/cjdb/constexpr-contracts
+Description: A constexpr-friendly, optimisation-friendly contracts library.
```
Comment (NancyLi1013) [inline ports/constexpr-contracts/CONTROL:4]: If the port doesn't support linux, please add `Supports: !linux ` in CONTROL file.

And also add `vcpkg_fail_port_install(ON_TARGET "linux")` at the top of portfile.cmake.

File: scripts/ci.baseline.txt [hunk 2]
```diff
@@ -227,6 +227,7 @@ coin:arm64-windows=fail
 coin:arm-uwp=fail
 coin:x64-uwp=fail
 collada-dom:x64-windows-static=fail
+constexpr-contracts:x64-linux=fail
```
Comment (NancyLi1013) [inline scripts/ci.baseline.txt:230]: If we add`Supports `field in CONTROL, there is no need to set the unsupported triplet as fail in ci.baseline.txt.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/constexpr-contracts/CONTROL [hunk 3]
```diff
@@ -0,0 +1,4 @@
+Source: constexpr-contracts
+Version: 2020-05-25
```
Comment (NancyLi1013) [inline ports/constexpr-contracts/CONTROL:2]:
```suggestion
Version: 2020-08-09
```

The version should be a date that you accept the commit. Please see

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/constexpr-contracts/CONTROL (hunk 1 above)
Comment (cjdb) [reply ports/constexpr-contracts/CONTROL:4]: The package supports Linux; but requires GCC 10, which the CI does not appear to support. I won't add anything that prohibits Linux usage, given that I can (and do) use it on WSL.

Comment (NancyLi1013) [reply ports/constexpr-contracts/CONTROL:4]: Thanks for your clarification.

**Review by NancyLi1013 — APPROVED** [review]

File: ports/constexpr-contracts/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO cjdb/constexpr-contracts
+    REF 58154e9010cb80aad4e95ef6f1835ebd7db3780a # commit 2020-05-25
+    SHA512 b634267a4044cd712c8e52f65cd305f437864cab591f2b22104581f70b305ba52889dd46724e6047386463a010ee78fdd951411ea3691b5725d52d13f7adda76
+    HEAD_REF master
+)
+
+
```
Comment (ras0219) [inline ports/constexpr-contracts/portfile.cmake:11]:
```suggestion
```

File: ports/constexpr-contracts/portfile.cmake [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}
+    PREFER_NINJA
+)
+
+vcpkg_install_cmake()
+vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/constexpr-contracts TARGET_PATH share/constexpr-contracts)
```
Comment (ras0219) [inline ports/constexpr-contracts/portfile.cmake:18]:
```suggestion
vcpkg_fixup_cmake_targets(CONFIG_PATH lib/cmake/constexpr-contracts)
```

**Review by ras0219 — APPROVED** [review]
Thanks for the PR!

**Pull request merged**